### PR TITLE
make comments section populating from servlet json

### PR DESCRIPTION
**Pull request by jackdavidweber**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -19,14 +19,26 @@
...
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      comments = new ArrayList<>();
+      comments.add("not bad");
+      comments.add("could be better");
+      comments.add("really liked it!");
+
+      String json = new Gson().toJson(comments);
+
     response.setContentType("text/html;");
```
Comment (Danish21) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I believe you also have to change the content type to application/json.

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -30,3 +30,52 @@ function addRandomQuote() {
...
+
+/**
+ * Fetches comments from the servers and adds them to the DOM.
+ */
+function getCommentsold() {
+  fetch('/data').then(response => response.json()).then((comments) => {
+    // comments is an object, not a string, so we have to
+    // reference its fields to create HTML content
+    console.log(comments);
+    const commentsArr = JSON.parse(comments);
```
Comment (Danish21) [inline portfolio/src/main/webapp/script.js]: I believe you won't have to do this after you change the content type in the server file as response.json() should take care of it.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -30,3 +30,52 @@ function addRandomQuote() {
...
+
+
```
Comment (Danish21) [inline portfolio/src/main/webapp/script.js:48]: Remove extra new lines. I suggest keeping one new line at most.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -30,3 +30,52 @@ function addRandomQuote() {
...
+  const response = await fetch('/data');
+  const quote = await response.text();
+  document.getElementById('fetchTutorial').innerText = quote;
+}
+
+
+/**
+ * Fetches comments from the servers and adds them to the DOM.
+ */
+function getCommentsold() {
```
Comment (Danish21) [inline portfolio/src/main/webapp/script.js]: Nit:
Use camel case i.e. getCommentSold

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -30,3 +30,52 @@ function addRandomQuote() {
...
+    const commentsListElement = document.getElementById('comments');
+    commentsListElement.innerHTML = '';
+    commentsListElement.appendChild(createListElement(commentsArr[0]));
+    
+  });
+}
+
+async function getComments() {
+  const response = await fetch('/data');
+  const comments = await response.text();
```
Comment (Danish21) [inline portfolio/src/main/webapp/script.js]: Same set the correct content type i.e. application/json and response.json(). No need to use JSON.parse.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (g-harel) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: `application/json` would be more appropriate now!

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -30,3 +30,52 @@ function addRandomQuote() {
...
+
+  const commentsArr = JSON.parse(comments);
+
+  const commentsListElement = document.getElementById('comments');
+  commentsListElement.innerHTML = '';
+
+  for (var i = 0; i < commentsArr.length; i++) {
+      commentsListElement.appendChild(createListElement(commentsArr[i]));
+
```
Comment (Danish21) [inline portfolio/src/main/webapp/script.js]: Remove extra new line.

File: portfolio/src/main/webapp/index.html [hunk 7]
```diff
@@ -520,6 +522,15 @@ <h2 class="heading">My Github</h2>
                         </div><!--//ghfeed-->
                     </div><!--//section-inner-->
                 </aside><!--//section-->
+
+                <aside class="Comments aside section">
+                    <div class="section-inner shadow-sm rounded">
+                        <h2 class="heading">Comments!</h2>
+                        <button onclick="getComments()">Fetch comments!</button>
+                        <ul id="comments"></ul>
+                        </div><!--//ghfeed-->
```
Comment (g-harel) [inline portfolio/src/main/webapp/index.html:533]: Is this line extra? Don't see what it's closing.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 8]
```diff
@@ -19,14 +19,26 @@
...
+
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    private List<String> comments;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      comments = new ArrayList<>();
```
Comment (Danish21) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: Nit: Make sure all statements within the scope use consistent indentation.

File: portfolio/src/main/webapp/script.js [hunk 9]
```diff
@@ -30,3 +30,52 @@ function addRandomQuote() {
...
+async function getComments() {
+  const response = await fetch('/data');
+  const comments = await response.text();
+
+  const commentsArr = JSON.parse(comments);
+
+  const commentsListElement = document.getElementById('comments');
+  commentsListElement.innerHTML = '';
+
+  for (var i = 0; i < commentsArr.length; i++) {
```
Comment (Danish21) [inline portfolio/src/main/webapp/script.js]: Nit: Use let instead of var.

File: portfolio/src/main/webapp/script.js [hunk 10]
```diff
@@ -30,3 +30,52 @@ function addRandomQuote() {
...
+
+
+/**
+ * Fetches comments from the servers and adds them to the DOM.
+ */
+function getCommentsold() {
+  fetch('/data').then(response => response.json()).then((comments) => {
+    // comments is an object, not a string, so we have to
+    // reference its fields to create HTML content
+    console.log(comments);
```
Comment (Danish21) [inline portfolio/src/main/webapp/script.js]: Remove console.log before merging.

File: portfolio/src/main/webapp/script.js [hunk 11]
```diff
@@ -30,3 +30,52 @@ function addRandomQuote() {
   const quoteContainer = document.getElementById('quote-container');
   quoteContainer.innerText = quote;
 }
+
+console.log("hello world");
```
Comment (Danish21) [inline portfolio/src/main/webapp/script.js]: Remove console.log before merging.

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (Danish21) [inline portfolio/src/main/webapp/script.js]: Is this need since you have getComments() below?

**Review by Danish21 — APPROVED** [review]

**Pull request merged**